client/dashboard/app/queries/purchase.ts

```diff
@@ -0,0 +1,7 @@
+import { mutationOptions } from '@tanstack/react-query';
+import { removePurchase } from '../../data/purchase';
+
+export const removePurchaseMutation = () =>
```

Comment:
For consistency, we usually add the parameter here instead of relying on the data layer. 


```suggestion
export const removePurchaseMutation = ( purchaseId: number ) =>
```

The other benefit is that if we later want to invalidate some queries, the parameter is already there.

Reply:
Sure, good call. I'll add it, but I'm not sure if it's a number or a string?

Reply:
I'm not sure, use whatever that works 😄

Reply:
Currently, the `purchaseId` is a parameter of the mutate function. The mutation parameter approach would create unnecessary complexity and performance overhead in a scenario where we have a screen with a list of purchases, where each item has a delete action, for example.

Reply:
> but I'm not sure if it's a number or a string?

Right in time: https://github.com/Automattic/wp-calypso/pull/105344 😆

Reply:
> The mutation parameter approach would create unnecessary complexity and performance overhead in a scenario where we have a screen with a list of purchases, where each item has a delete action, for example.

Interesting... OK let's do that for now. I'm interested to see how this pattern allows query invalidation later if needed 🤔